remix/app/routes/_app.newsletter.create.test.tsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * @jest-environment node
+ */
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { Newsletter } from '@prisma/client';
+
+import { action } from './_app.newsletter.create';
+import { authenticator } from '../config/auth.server';
+import { prismaMock } from '../tests/database';
+import { newsletterFactory } from '../tests/factories/newsletter.factory';
+import { userFactory } from '../tests/factories/user.factory';
+import { makeRequest } from '../tests/helpers/request';
+
+jest.mock('../config/auth.server', () => ({
+  authenticator: {
+    isAuthenticated: jest.fn(),
+  },
+}));
+
+describe('POST /newsletter/create', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const user = { ...userFactory };
+  const newsletter = { ...newsletterFactory };
+
+  (authenticator.isAuthenticated as jest.Mock).mockResolvedValue(user);
+
+  describe('givens a newsletter data', () => {
+    it('creates a newsletter', async () => {
+      prismaMock.user.findUnique.mockResolvedValue(user);
+      prismaMock.newsletter.create.mockResolvedValue(newsletter);
```

Comment:
I think it will be a little bit better if we mock the [`NewsletterRepository.create`](https://github.com/nimblehq/js-framework-benchmark/pull/112/files#diff-d5b82c718c7d3096aff7681c92400168efc740cbdf51d42d1cfe2c496b4e453eR31) instead of `prismaMock.newsletter.create`, since we are using `NewsletterRepository.create` instead of `prismaMock.newsletter.create` in file `remix/app/routes/_app.newsletter.create.tsx`

Reply:
In my opinion, leave it like this should be ok, as `NewsletterRepository` is a class that define `Prisma` method.
And mock the `Prisma` might make the testing development feel simpler.

Reply:
Thank you for clarify me that @olivierobert.